### PR TITLE
LPS-151416 Select's field predefined value is not respected if option is changed or deleted

### PR DESCRIPTION
**Pull request by diogomelo09**
[LPS-151416](https://issues.liferay.com/browse/LPS-151416)

### CODE REVIEW

**Comment by diogomelo09** [comment]
ci:test:sf

**Comment by diogomelo09** [comment]
ci:test:relevant

**Comment by michaelmdelcavalcanti** [comment]
ci:test:relevant

**Comment by diogomelo09** [comment]
ci:test:relevant

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-web/src/main/resources/META-INF/resources/js/ddm_form.js [hunk 1]
```diff
@@ -3816,7 +3815,23 @@ AUI.add(
...
+					if (count === value.length) {
+						const name = instance.get('name');
+						const field = instance.getFieldByNameInFieldDefinition(
+							name
+						);
+						value =
+							field.predefinedValue[
+								instance.get('displayLocale')
+							];
+					}
```
Comment (cgmonte) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-web/src/main/resources/META-INF/resources/js/ddm_form.js:3834]: Hi @diogomelo09, thanks for sending this PR! I think we might not be covering the cases where there are multiple selectable options, multiple predefined values and multiple missing values here.

Please try the following and see if it works better:

```
setValue(values) {
	const instance = this;
	const currentlyAvailableValues = instance
		.getInputNode()
		.all('option')
		.val();
	const predefinedValues = JSON.parse(
		instance.getFieldByNameInFieldDefinition(
			instance.get('name')
		).predefinedValue[instance.get('displayLocale')]
	);

	if (Lang.isString(values)) {
		if (values !== '') {
			values = JSON.parse(values);
		} else {
			values = [''];
		}
	}

	let resultValues = [];

	values.forEach((value) => {
		if (!currentlyAvailableValues.includes(value)) {
			resultValues.push(...predefinedValues);
		} else {
			resultValues.push(value);
		}
	})
	
	resultValues = [...new Set(resultValues)];

	instance
		.getInputNode()
		.all('option')
		.each((item) => {
			item.set(
				'selected',
				resultValues.indexOf(item.val()) > -1
			);
		});
},
```
let me know if you need help :)

**Review by cgmonte — CHANGES_REQUESTED** [review]
Please take a look at he suggested changes, thanks a lot!

**Review by cgmonte — APPROVED** [review]
LGTM!

**Comment by cgmonte** [comment]
ci:forward

**Comment by cgmonte** [comment]
Hey @diogomelo09, there was a [poshi failure](https://test-1-38.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/349235/testReport/com.liferay.poshi.runner/PoshiRunner/test_LocalFile_FormsSelectFromListField_CanAllowMultipleSelections_/) that might be related to the changes in this PR. Can you take a look? You can talk to a QA engineer if you need help to analyse the failure.

**Comment by cgmonte** [comment]
ci:forward

**Pull request closed without merging**